core/src/services/s3/backend.rs

```diff
@@ -1325,10 +1329,30 @@ impl S3Backend {
         Ok(req)
     }
 
-    fn s3_get_object_request(&self, path: &str, range: BytesRange) -> Result<Request<AsyncBody>> {
+    fn s3_get_object_request(
+        &self,
+        path: &str,
+        range: BytesRange,
+        override_content_disposition: Option<&str>,
+    ) -> Result<Request<AsyncBody>> {
         let p = build_abs_path(&self.root, path);
 
-        let url = format!("{}/{}", self.endpoint, percent_encode_path(&p));
+        // Construct headers to add to the request
+        let mut query_args = Vec::new();
+        if let Some(override_content_disposition) = override_content_disposition {
+            query_args.push(format!(
+                "{}={}",
+                constants::RESPONSE_CONTENT_DISPOSITION,
+                percent_encode_path(override_content_disposition)
+            ))
+        }
+
+        let mut url = format!("{}/{}", self.endpoint, percent_encode_path(&p));
```

Comment:
I prefer to simplify them by directly writing the string.

Reply:
Im not sure I understand what you mean. Do you mean not adding the query string args to a vec first? How would you then add the logic to add the `?` and the `&` in between subsequent query arguments? Or did I over-engineer it already by adding support for adding multiple query string arguments? 😅

Reply:
> Or did I over-engineer it already by adding support for adding multiple query string arguments?

Aha, I mean this. we can simply add a `?xxx=yyy` after the url.

Reply:
Done!